tasks/grunt-postmark-templates.js

```diff
@@ -0,0 +1,146 @@
+    var options = this.options();
+    var template = this.data;
+
+    var serverToken = options.serverToken || grunt.config('secrets.serverToken') || grunt.config('secret.postmark.server_token');
+
+    if (!serverToken) {
+      grunt.fail.warn('Missing required option "serverToken" \n');
+    }
+
+    template.name = template.name || this.target;
+
+    if (!template.name) {
+      grunt.fail.warn('Missing required template property "name" \n');
+    }
+
+    if (!template.subject) {
+      grunt.fail.warn('Missing required template property "subject" \n');
+    }
+
+    var postmark = require('postmark');
+    var client = new postmark.Client(serverToken);
+
+    // read the referenced files, but hold on to the original filenames
+    var expanded = Object.assign({}, template);
+
+    if (expanded.htmlBody) {
+      expanded.htmlBody = grunt.file.read(expanded.htmlBody);
+    }
+    if (expanded.textBody) {
+      expanded.textBody = grunt.file.read(expanded.textBody);
+    }
+
+    if (template.templateId) {
+      client.editTemplate(template.templateId, expanded, function(err, response) {
+        if (err && err.code === 1101) {
+          grunt.log.warn('Template ' + template.templateId + ' not found, so attempting create');
+          delete template.templateId;
+          delete expanded.templateId;
+          client.createTemplate(expanded, function(err, response) {
+            handleResponse(template, err, response, done);
+          });
+        } else {
+          handleResponse(template, err, response, done);
+        }
+      });
+    } else {
+      client.createTemplate(expanded, function(err, response) {
+        handleResponse(template, err, response, done);
+      });
+    }
+
+  });
+
+  function handleResponse(template, err, response, done) {
+    if (err){
+      errorMessage(err);
+      done();
+    } else {
+      template.templateId = response.TemplateId;
+      // append this record to the result array
+      var upd = grunt.config.get('updatedTemplates');
+      var tname = template.name;
+      delete template.name;
+      upd[tname] = template;
+      grunt.config.set('updatedTemplates', upd);
+
+      successMessage(tname, template.templateId);
+      done(template);
+    }
+  }
+
+  function errorMessage(err) {
+    if (err.message) {
+      grunt.log.warn('Error: ' + err.message);
+    } else {
+      grunt.log.warn('Error: ' + JSON.stringify(err));
+    }
+  }
+
+  function successMessage(name, templateId) {
+    grunt.log.writeln('Template ' + name + ' pushed: ' + JSON.stringify(templateId));
+  }
+
+  grunt.registerTask('_postmark-output', 'writes out the resulting template IDs', function() {
+
+    var options = this.options({
+      filename: "templates-output.json"
+    });
+
+    var results = grunt.config('updatedTemplates');
+
+    grunt.file.write(options.filename, JSON.stringify(results, null, 2));
+
+    grunt.log.writeln("Updated template information written to " + options.filename);
```

Comment:
Circling back and we shouldn't worry about this for now. Adding a note to the documentation should be good enough.